core/ecschema-editing/src/Merging/SchemaMerger.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Schema, SchemaItemType } from "@itwin/ecschema-metadata";
+import { SchemaChanges, SchemaItemChanges } from "../Validation/SchemaChanges";
+import { SchemaComparer } from "../Validation/SchemaComparer";
+
+import mergeClasses from "./ClassMerger";
+import mergeEnumeration from "./EnumerationMerger";
+import mergePropertyCategory from "./PropertyCategoryMerger";
+import mergeSchemaItems from "./SchemaItemMerger";
+import mergeSchemaReferences from "./SchemaReferenceMerger";
+import mergeCAClasses from "./CAClassMerger";
+
+/**
+ * Defines the context of a Schema merging run.
+ * @beta
+ */
+export interface SchemaMergeContext {
+  readonly targetSchema: Schema;
+  readonly sourceSchema: Schema;
+}
+
+/**
+ * Class to merge two schemas together.
+ * See @see SchemaMerger.merge for details.
```

Comment:
```suggestion
 * @see [[merge]] to merge the schemas.
```

Though it's unclear to me why users of this API must instantiate a stateless instance of a class with one method to merge two schemas.

`const schema = new SchemaMerger().merge(target, src);`
vs
`const schema = mergeSchemas(target, src);`

Reply:
The class will get more methods or properties in future, to be able to configure the merging workflow.